python/valis/routes/target.py

```diff
@@ -182,6 +183,7 @@ async def get_target_altid(self,
     @router.get('/spectra/{sdss_id}', summary='Retrieve a spectrum for a target sdss_id',
                 dependencies=[Depends(get_pw_db)],
                 response_model=List[SpectrumModel])
+    @valis_cache()
     async def get_spectrum(self, sdss_id: Annotated[int, Path(title="The sdss_id of the target to get", example=23326)],
                            product: Annotated[str, Query(description='The file species or data product name', example='specLite')],
```

Comment:
The return response in `target/spectrum` route calls `get_a_spectrum`, which was a generator.  Can you check if this route works as intended with the cache?  or see if we need to adjust the underlying function?

Reply:
This seems to work fine for me, but the example request

```console
curl -X 'GET' \
        'http://127.0.0.1:8000/target/spectra/23326?product=specLite&ext=BOSS%2FAPO&release=DR18' \
        -H 'accept: application/json'
```

return an empty list and I'm not sure how to find something that will return a full spectrum. Can you check this one?

BTW, this route fails if you just `poetry install` valis with error

```python
  ...
  File "/Users/gallegoj/Documents/Code/sdss5/valis/python/valis/utils/versions.py", line 97, in get_tags
    return get_latest_tag_info() if release == 'WORK' else get_tag_info(release)
  File "/Users/gallegoj/Documents/Code/sdss5/valis/python/valis/utils/versions.py", line 33, in get_tag_info
    raise RuntimeError('No tag models found.')
RuntimeError: No tag models found.
```

It seems one needs to manually clone and pip install the `datamodel` for this to work.

Reply:
For local use, you need the file locally.  Otherwise it returns nothing.   

And yeah, some routes need the datamodel product installed locally.  I didn't want to make the datamodel package a strict dependency of valis.  Most of valis only needs the datamodel for its basic python access, but some of the `info` routes need the full product list.  It may be worth either splitting the datamodel into two products, or producing a lightweight pip installable package.  But for now yeah you need to git clone and manually install.

Reply:
For the both the spectrum and pipelines route, I get the following error, with a longer traceback that reaches into the cache encoder.   I tried both a `null` and `in-memory` backend and get the same error.  

spectrum route `http://localhost:8000/target/spectra/23326?product=specLite&ext=BOSS%2FAPO&release=IPL3`
```
INFO:     127.0.0.1:65300 - "GET /target/spectra/23326?product=specLite&ext=BOSS%2FAPO&release=IPL3 HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/brian/anaconda3/envs/valis_solara/lib/python3.10/site-packages/fastapi/encoders.py", line 322, in jsonable_encoder
    data = dict(obj)
TypeError: cannot convert dictionary update sequence element #0 to a sequence

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/brian/anaconda3/envs/valis_solara/lib/python3.10/site-packages/fastapi/encoders.py", line 327, in jsonable_encoder
    data = vars(obj)
TypeError: vars() argument must have __dict__ attribute

The above exception was the direct cause of the following exception:
```
pipelines route `http://localhost:8000/target/pipelines/23326?pipe=all&release=IPL3`
```
INFO:     127.0.0.1:49185 - "GET /target/pipelines/23326?pipe=all&release=IPL3 HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/brian/anaconda3/envs/valis_solara/lib/python3.10/site-packages/fastapi/encoders.py", line 322, in jsonable_encoder
    data = dict(obj)
ValueError: dictionary update sequence element #0 has length 1; 2 is required

During handling of the above exception, another exception occurred:

```

Reply:
That didn't work.  The pipelines endpoint doesn't return an iterator but returns a similar error.   I'll also try to look into it.

Reply:
I think the issue was the default coder for the cache couldn't serialize some of the outputs from the spectrum and pipelines routes, e.g. numpy data or python memoryview objects.  And the fact that the cache serializes the entire response before it gets parsed through any FastAPI ResponseModel, which I often use to only return a subset of info, or ignore problem fields.

I added a custom orjson encoder for the cache, with a special handler for dealing with python memoryview objects.  For reference, the `sdss5_targeting_flags` field returned for `boss` and `astra` output in the pipelines route.  

These routes now load properly and the Zora target page renders successfully.

Reply:
Great, that makes sense. I'm not surprised that the serialisation of the response for the cache object is less robust against things like numpy arrays and memoryviews than the pydantic model.